### PR TITLE
Update use_stm8dce.py

### PR DESCRIPTION
**Pull request by ghost**
new

### CODE REVIEW

File: use_stm8dce.py [hunk 1]
```diff
@@ -23,33 +24,31 @@
...
-    stm8dce_flags=   env.GetProjectOption("stm8dce_flags", default="") 
-    
     asm_path=""
 
     for x in source:
       tst_asm_pth= os.path.splitext(str(x))[0]+".asm"  
       if  os.path.isfile(tst_asm_pth):
         asm_path += ' "' + tst_asm_pth +'"'
-
+
```
Comment (maxgerhardt) [inline use_stm8dce.py]: Random addition of two spaces is weird.

File: use_stm8dce.py [hunk 2]
```diff
@@ -23,33 +24,31 @@
...
+    
     for x in source:
         tmp_path= os.path.join(temp_out_dir,os.path.splitext(os.path.basename(str(x)))[0]+".asm") 
         if  os.path.isfile(tmp_path):
-            env.Execute(env.VerboseAction("$AS -plosg -ff -o " + '"' +str(x) +'" "' +tmp_path+'"',"COMPILING "+str(x) ))
-            os.remove(tmp_path)
-            
-    if not os.listdir(temp_out_dir):
-        os.rmdir(temp_out_dir)
+            env.Execute(env.VerboseAction("$AS -plosg -ff -o " + '"' +str(x) +'" "' +tmp_path+'"',"RECOMPILING "+str(x) ))
```
Comment (maxgerhardt) [inline use_stm8dce.py]: Technically this is "Reassembling" the optimized files, we are invoking the assembler (`$AS`) too.

File: use_stm8dce.py [hunk 3]
```diff
@@ -23,33 +24,31 @@
...
   
+    
     for x in source:
         tmp_path= os.path.join(temp_out_dir,os.path.splitext(os.path.basename(str(x)))[0]+".asm") 
         if  os.path.isfile(tmp_path):
-            env.Execute(env.VerboseAction("$AS -plosg -ff -o " + '"' +str(x) +'" "' +tmp_path+'"',"COMPILING "+str(x) ))
-            os.remove(tmp_path)
-            
-    if not os.listdir(temp_out_dir):
-        os.rmdir(temp_out_dir)
```
Comment (maxgerhardt) [inline use_stm8dce.py:53]: So the temporary output directory is never removed after usage?

**Review by maxgerhardt — CHANGES_REQUESTED** [review]

File: use_stm8dce.py (hunk 2 above)
Comment (ghost) [reply use_stm8dce.py]: ok

File: use_stm8dce.py (hunk 1 above)
Comment (ghost) [reply use_stm8dce.py]: I checked everything is ok

File: use_stm8dce.py (hunk 3 above)
Comment (ghost) [reply use_stm8dce.py:53]: You can leave this folder, it does not affect the project build in any way, if it is deleted via Clean

File: use_stm8dce.py [hunk 4]
```diff
@@ -29,9 +29,9 @@ def optimize_asm(source, target, env):
     asm_path=""
     
     for x in source:
-      tst_asm_pth= os.path.splitext(str(x))[0]+".asm"  
-      if  os.path.isfile(tst_asm_pth):
-        asm_path +=' "'+tst_asm_pth +'"'
+        tst_asm_pth= os.path.splitext(str(x))[0]+".asm"  
+        if  os.path.isfile(tst_asm_pth):
+            asm_path +=' "'+tst_asm_pth +'"'
```
Comment (maxgerhardt) [inline use_stm8dce.py]: This made the indentation spaces worse than before -- too many.

Please run a Python code formatter on the whole thing once.

**Review by maxgerhardt — CHANGES_REQUESTED** [review]

**Pull request merged**